### DISCUSSION TITLE
Common C Library for z/OS ports

### DISCUSSION THREAD

**Discussion by IgorTodorovskiIBM** [zopencommunity/meta#37] [General]
Why should we consider introducing a common library for z/OS ports?

As we continue to port tools to z/OS, we continue to encounter the following issues:

---
  * **Problem**: We need to identify where new file descriptors are created as newly created file descriptors need to be tagged to support autotagging in enhanced ascii mode
  * **Solution**: Use a library to overrides LE open/pipe and tag the new file descriptors without changes to the project source code
---
  * **Problem**: Implement or workaround missing Linux/POSIX functions in LE
  * **Solution**: Write it once in a library and expose it to all other z/OS ports
--- 
  * **Problem**: Many of the ports have their own solution for how to handle ASCII/EBCDIC I/O, especially how untagged files are handled
  * **Solution**: Write a library to handle ASCII/EBCDIC considerations and introduce consistent environment variables 
---
**Idea:**
* Each project will Introduce a dependency to this common library.  This would mean adding the -I option to find the headers, the -L option to find the libraries and the -l option to link to the library/sidedeck
* As we upstream our ports, we need to make sure that the various open source communities are willing to adopt the library.  

**Recommendation:**
* Leverage an existing library like [zoslib](https://github.com/ibmruntimes/zoslib) which is currently being used for the Node.js on z/OS project.

Any thoughts? other suggestions? Let us know!

**Comment by MikeFultonDev** [comment]
done. we use zoslib
